kyo-core/shared/src/main/scala/kyo/Hub.scala

```diff
@@ -165,33 +249,45 @@ object Hub:
       *   The result of applying the function
       */
     def initWith[A](capacity: Int)[B, S](f: Hub[A] => B < S)(using Frame): B < (S & IO) =
-        Channel.initWith[A](capacity) { ch =>
-            IO {
-                val listeners = new CopyOnWriteArraySet[Channel[A]]
-                Async.run {
-                    Loop.foreach {
-                        ch.take.map { v =>
-                            IO {
-                                val puts =
-                                    listeners.toArray()
-                                        .toList.asInstanceOf[List[Channel[A]]]
-                                        .map(child => Abort.run[Throwable](child.put(v)))
-                                Async.parallelUnbounded(puts).map(_ => Loop.continue)
-                            }
+        IO.Unsafe {
+            val channel          = Channel.Unsafe.init[A](capacity, Access.MultiProducerSingleConsumer).safe
```

Comment:
Using relaxed access policy for both the main and listener buffers since it provides better performance. Since there's only one fiber handling the distribution, the global buffer is `MultiProducerSingleConsumer` and the listener buffer is `SingleProducerMultiConsumer`